WalletWasabi/WabiSabi/Client/OwnershipProofValidator.cs

```diff
@@ -0,0 +1,127 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Crypto;
+using WalletWasabi.Stores;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	// OwnershipProofValidator validates the ownership proofs provided by the coordinator.
+	public class OwnershipProofValidator
+	{
+		public OwnershipProofValidator(TransactionStore transactionStore, IBlockProvider blockProvider)
+		{
+			TransactionStore = transactionStore;
+			BlockProvider = blockProvider;
+		}
+
+		public TransactionStore TransactionStore { get; }
+		public IBlockProvider BlockProvider { get; }
+
+		// Verifies the coins and ownership proofs. These proofs are provided by the coordinator who should
+		// have validated them before which means that when dealing with an honest coordinator all these proofs
+		// are valid.
+		//
+		// In case one proof is invalid we have evidence enough to know the coordinator is malicious and we
+		// have to abort the process, notify the user and ban the coordinator.
+		public async ValueTask<int> VerifyOtherAlicesOwnershipProofsAsync(
+			uint256 roundId,
+			IEnumerable<(uint256 BlockId, OutPoint OutPoint, Money Amount, OwnershipProof OwnershipProof)> othersCoins,
+			int minimumNumberOfValidProofs,
+			CancellationToken cancellationToken)
+		{
+			var validProofs = 0;
+			var coinJoinInputCommitmentData = new CoinJoinInputCommitmentData("CoinJoinCoordinatorIdentifier", roundId);
+
+			var mineTxIds = TransactionStore.GetTransactionHashes();
```

Comment:
```suggestion
			var myConfirmedTxIds = TransactionStore.GetTransactionHashes();
```

repeating previous feedback, i think naming it like this will eliminate confusion between "mined" and "my" by clearly indicating it's both

Reply:
ACK